plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -215,13 +221,7 @@ emptyWalletState w = WalletState pk 0 oa mempty where
 
 -- | Events produced by the blockchain emulator.
 data EmulatorEvent =
-    TxnSubmit TxId
-    -- ^ A transaction has been added to the pool of pending transactions.
-    | TxnValidate TxId
-    -- ^ A transaction has been validated and added to the blockchain.
-    | TxnValidationFail TxId Index.ValidationError
-    -- ^ A transaction failed  to validate.
-    | SlotAdd Slot
+    BlockProcessed [ServerEvent]
```

Comment:
This doesn't seem like the right name - I'm assuming we may get events at other times! I'd probably just have `ServerEvent ServerEvent` or something.